### PR TITLE
For the cronjob, avoid re-downloading the cache dataset if present

### PR DESCRIPTION
**Pull request by h-mayorquin**
There is a new option in the cache machinery that allows for only checking if the cache is available and not downloading it:

https://github.com/actions/cache#inputs

This will save some unecessary computation and bandwith use and as an important side effect make @samuelgarcia happier.

This is used in the daily cron job to checks if it is necessary to create new caches for both the gin data and pip dependencies.

### CODE REVIEW

**Comment by alejoe91** [comment]
Thanks @h-mayorquin! If I understand correctly, this additional argument will skip recreating a cache for datasets if that's already present. Right?

**Comment by h-mayorquin** [comment]
@alejoe91 yes, you are correct, the job of the cron job here is to download the datasets if they are not available. That is, in the cronjobs the dataset is not actually used so we don't need to download it. This new option they recently added implements that. Saving behavior is not affected so if the data is not available it will be created and uploaded as it should.

File: .github/workflows/caches_cron_job.yml [hunk 1]
```diff
@@ -33,6 +33,7 @@ jobs:
         with:
           path: ${{ github.workspace }}/test_env
           key: ${{ runner.os }}-venv-${{ steps.dependencies.outputs.hash }}-${{ steps.date.outputs.date }}
+          lookup-only: 'true'
```
Comment (h-mayorquin) [inline .github/workflows/caches_cron_job.yml:36]:
```suggestion
          lookup-only: 'true'   # Avoids downloading the data, saving behavior is not affected.
```

File: .github/workflows/caches_cron_job.yml [hunk 2]
```diff
@@ -64,6 +65,7 @@ jobs:
         with:
           path: ~/spikeinterface_datasets
           key: ${{ runner.os }}-datasets-${{ steps.repo_hash.outputs.dataset_hash }}
+          lookup-only: 'true'
```
Comment (h-mayorquin) [inline .github/workflows/caches_cron_job.yml:68]:
```suggestion
          lookup-only: 'true'   # Avoids downloading the data, saving behavior is not affected.
```

**Comment by alejoe91** [comment]
Let's merge after the release, since it will only affect cron jobs ;)

**Comment by h-mayorquin** [comment]
No rush here.

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**